src/Core/Settings.h

```diff
@@ -38,6 +38,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
   */
 
 #define COMMON_SETTINGS(M) \
+    M(String, sql_dialect, "clickhouse", "Which SQL dialect will be used to parse query", 0)\
```

Comment:
let's make it just "dialect"

Reply:
This feature looks quite marginal, so I would prefer to use longer setting name. We could use just `dialect` setting of type `Enum` (not `String`), but we don't have other dialects for now (and I'm not sure we will have more dialects in the near future). What do you think about something like `use_kusto_query_language_dialect` of type `Bool`?

Reply:
@tavplubix `Bool` will exclude any other dialects in the future if we decide to add any other - it should be mutually exclusive...

Reply:
there is another PR use the dialect.:
https://github.com/ClickHouse/ClickHouse/pull/36308/files
so there will be more dialects

Reply:
Ok, so it makes sense to name it just `dialect`, but it must be `Enum`, not `String`

Reply:
we changed the dialect  to Enum.